src/services/schedule.js

```diff
@@ -252,37 +245,21 @@ const ScheduleService = () => {
         return [currentTask].concat(solve(restTasks, freeIntervals));
       }
 
-      // Go through all branches and find max number of tasks scheduled
-      let maxCount = 0;
-      // console.log('\nTask: ' + currentTask.id);
-      // console.log('Branches: ');
-      for (branch of allBranches) {
-        let count = 0;
-        // console.log(branch);
-        for (task of branch) {
-          if (task.scheduled_time != null) {
-            count++;
-          }
-        }
-        maxCount = count > maxCount ? count : maxCount;
-      }
+      // Go through all branches and get scores
+      let scores = [];
 
-      // Choose solution with max number
       for (branch of allBranches) {
-        let count = 0;
+        let score = 0;
         for (task of branch) {
           if (task.scheduled_time != null) {
-            count++;
+            score +=
+              (getUnix(task.due_date) - task.scheduled_time) / task.duration;
```

Comment:
This assigns a higher score to tasks that are short and scheduled much earlier than their due date.